soccer/radio/Radio.hpp

```diff
@@ -23,15 +26,31 @@ class Radio {
 
     int channel() const { return _channel; }
 
-    const std::vector<Packet::RadioRx>& reversePackets() const {
-        return _reversePackets;
+    bool hasReversePackets() {
+        std::lock_guard<std::mutex> lock(_reverse_packets_mutex);
+        return _reversePackets.size();
     }
 
-    std::vector<Packet::RadioRx>& reversePackets() { return _reversePackets; }
+    inline const Packet::RadioRx popReversePacket() {
+        std::lock_guard<std::mutex> lock(_reverse_packets_mutex);
+        Packet::RadioRx packet = std::move(_reversePackets.front());
+        _reversePackets.pop();
+        return packet;
+    }
 
-    void clear() { _reversePackets.clear(); }
+    void clear() {
+        std::lock_guard<std::mutex> lock(_reverse_packets_mutex);
+
+        // Clear the reverse packets queue by copy-swap
+        std::queue<Packet::RadioRx> cleared;
```

Comment:
Any reason why you want to use a swap instead of just clear?

Reply:
std::queue doesn't have clear. You could use deque but I don't think it matters

Reply:
Hey Jay! Is there a reason we're using a queue? It just wraps a queue. Anyways. You can make this simpler and save a copy by saying _reverse_packets = std::queue<Packet::RadioRx>{}; Or probably just _reverse_packets={};

Reply:
`_reverse_packets = {}` breaks the CI, and this copy-swap is a well-known C++ idiom that in pretty sure is going to end up being optimized away (maybe I'm overly optimistic about the optimizer's quality, though). Probably better to replace it with a dequeue anyway.